internal/discovery/discovery.go

```diff
@@ -248,6 +250,12 @@ func (c *DiscoveredConfig) Parse(ctx context.Context, l log.Logger, opts *option
 		config.ExcludeBlock,
 	)
 
+	if suppressParseErrors {
+		// If suppressing parse errors, we want to avoid writing diagnostics to the log.
+		parseOptions := append(parsingCtx.ParserOptions, hclparse.WithDiagnosticsWriter(io.Discard, true))
+		parsingCtx = parsingCtx.WithParseOption(parseOptions)
```

Comment:
What if the error is actually important? Shouldn't we try to filter diagnostics?

My concern is that something like an HCL function fails, preventing parse of something important in the dependency/dependent and the user doesn't know why it isn't working.